### PR TITLE
feat: Update balance calculation with 3-tier fallback

### LINKED ISSUE #27: Holdings Value: Update balance calculation with 3-tier fallback
## Overview
Part of the approved Holdings Value Calculation design.

## Changes Required
Update `getBalanceData` to use fallback order:
1. **Account Snapshot** (highest priority)
2. **Holding Snapshot** (medium priority) - calculated from holdings value
3. **Transactions** (lowest priority)

### New Functions
- `getBalanceDataFromHoldingSnapshots()` - Calculate balance from holding snapshots + security prices

### Modified Functions
- `getBalanceData()` - Merge three sources with priority order

## Files
- `src/client/lib/hooks/calculation/accounts.ts`

## Related
- Design doc: https://doc.hoie.kim/p/ZDhNQZtVic

### PR DESCRIPTION
**Pull request by moltboie**
## Summary
Implements Holdings Value issue #27 - Update balance calculation with 3-tier fallback.

## Changes Required
Update `getBalanceData` to use fallback order:
1. **Account Snapshot** (highest priority) - direct balance from Plaid snapshots
2. **Holding Snapshot** (medium priority) - calculated from sum of holding values
3. **Transactions** (lowest priority) - derived from transaction history

## New Functions
- `getBalanceDataFromHoldingSnapshots()` - Calculate balance from holding snapshots by summing `institution_value` for all holdings per account per period

## Modified Functions
- `getBalanceData()` - Now accepts `holdingSnapshots` parameter and merges three sources with priority order

## Files Changed
- `src/client/lib/hooks/calculation/accounts.ts` - Main implementation
- `src/common/models/Account.ts` - Add `useHoldingSnapshots` to `AccountGraphOptions`
- `src/client/lib/models/Account.ts` - Default `useHoldingSnapshots: true`
- `src/server/lib/postgres/models/account.ts` - Add DB column
- `src/server/lib/postgres/models/common.ts` - Add column constant
- `src/client/components/App/lib/hooks.ts` - Pass holdingSnapshots to getBalanceData
- `src/client/components/ConnectionProperties/index.tsx` - Include useHoldingSnapshots
- `src/server/lib/plaid/accounts.ts` - Include useHoldingSnapshots in defaults
- `src/server/lib/simple-fin/translators.ts` - Include useHoldingSnapshots in defaults
- `src/client/lib/indexed-db/service.ts` - Fix existing bug (missing type parameter)
- `src/client/lib/hooks/calculation/accounts.test.ts` - New tests for getAccountBalance

## Testing
- All existing tests pass (125 tests)
- Added 6 new unit tests for `getAccountBalance` function
- Integration testing requires e2e tests due to Dictionary class limitations in test environment

## Database Migration
New column `graph_options_use_holding_snapshots` added to accounts table with DEFAULT TRUE. Existing installations will need to run a migration to add this column.

## Related
- Design doc: https://doc.hoie.kim/p/ZDhNQZtVic

Closes #27

### CODE REVIEW

**Review by moltboie — COMMENTED** [review]
## Self-Review

### Checked:
- **Logic**: 3-tier fallback (snapshot → holdings → transactions) correctly prioritizes data sources
- **Types**: `HoldingSnapshot[]` parameter properly typed, `useHoldingSnapshots` added to `AccountGraphOptions`
- **Edge cases**: Empty holdings array handled, undefined checks in place
- **Database**: New column `graph_options_use_holding_snapshots` with DEFAULT TRUE
- **Tests**: 6 new unit tests for `getAccountBalance` function

### E2E Testing:
- Started server with `bun run start` ✓
- Login works: `POST /api/login` returns success ✓
- App loads at http://localhost:3005 ✓
- Accounts page shows 12 accounts with balance data ✓
- Balance displayed correctly: $<balance> for <employer-equity-account> ✓
- Account detail shows graph with historical balance ✓
- New DB column verified: `graph_options_use_holding_snapshots = true`

### Issues found:
- None

### Confidence: High
Balance fallback logic working correctly. UI displays values as expected.


*[Edited 2026-09-15: real employer-equity account label and balance redacted]*

File: src/server/lib/plaid/accounts.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export const getAccounts = async (user: MaskedUser, items: JSONItem[]) => {
           custom_name: "",
           hide: false,
           label: { budget_id: null },
-          graphOptions: { useSnapshots: true, useTransactions: true },
+          graphOptions: { useSnapshots: true, useHoldingSnapshots: true, useTransactions: true },
```
[the remark this answers is not in the document]
Comment (moltboie) [reply src/server/lib/plaid/accounts.ts]: Created issue #118 to track this refactor. Agreed — external data layers should return only external data, and custom properties should be applied in a composition layer.

**Comment by moltboie** [comment]
Fixed in 578df5f:
- Changed test import from \`from "../../models/Account"\` to \`from "client"\`
- Also fixed \`call.ts\` to lazy-initialize \`window.caches\` so the module import works in test environment

You're right — I should have found this when addressing the earlier feedback. I've now scanned all changed files in this PR for relative imports and confirmed none remain. Will apply this discipline consistently going forward: when fixing an issue, check the whole PR for the same pattern.

**Review by hoiekim — APPROVED** [review]
